File: ecom_web/src/types/Product.ts
export interface Product {
    id: number;
    name: string;
    price: number;
    quantity: number;
    categoryId: number;
    imagePath: string;
}
export interface ProductFormData {
 
  name: string;
  price: number;
   quantity: number;
    categoryId: number;
  image?: File; 
}

File: backend/project1/Models/Product.cs
﻿using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace project1.Models
{
    public class Product
    {
        public int Id { get; set; }

        public required string Name { get; set; }
        public decimal Price { get; set; }
        
        public int Quantity { set; get; }
        public string? ImagePath{ get; set; }

        //Foreign key
        public int CategoryId { get; set; }

        


        //Navigation property for Category
        [JsonIgnore]
        public Category Category { get; set; }    = null!;
    }
}
